### PR TITLE
Filter fixes

### PR DESCRIPTION
**Pull request by rowanwins**
- [x] Fix up clearing the temporal extent filter
- [x] Fix up clearing collections field on global search 
- [x] Fix up ID support 

I did a check on the others and I think they are all ok but let me know if there was anything else @m-mohr 

Resolves https://github.com/radiantearth/stac-browser/issues/161

### CODE REVIEW

File: src/models/ItemSearchCore.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ export default class ItemSearchCore {
   getAsCql2Json () {
     let out = {};
     this.queryableInputs.forEach(q => {
-      if (q.props.value !== '') {
+      if (q.props.value !== '' || q.props.value.length !== 0) {
```
Comment (m-mohr) [inline src/models/ItemSearchCore.js:79]: Shouldn't this be:
```suggestion
      if (q.props.value !== '' || (Array.isArray(q.props.value) && q.props.value.length !== 0)) {
```

**Comment by m-mohr** [comment]
Removing the collection field in global search doesn't work for me. I get no results on MS PC after clearing the collections, while before I got tons of results.

**Comment by m-mohr** [comment]
I also just realized that adding an Item ID in global search doesn't work, it is not taken into account.

**Comment by rowanwins** [comment]
I've done a fairly big refactor of the filtering stuff, hopefully this should address #180 and #162 . 

There is however one issue going on where the ItemFilter thinks I'm attempting to mutate something in the vuex store after the filter has been initially set. I think due to some behaviour around `this.filter` either being something local or something stored in the vuex store from the watcher in the ItemFilter - although it's not clear to me how the other variables (like `this.filter.datetime`) around getting around this?

**Comment by m-mohr** [comment]
Thanks for the refactor. 👍  I'll look into it this week and will also check the mutation issue.

**Comment by rowanwins** [comment]
> PPS: This solves https://github.com/radiantearth/stac-browser/issues/162 only partially, it does not solve the $ref issue.

From what I can tell the ref part of things is working ok, for example the CloudCover field comes via the refs.

**Comment by m-mohr** [comment]
Oh, yeah, indeed. That was an oversight from my side, $refs seem to work. 

Another issue found is that if you open a collection filter first and then open search, it will show the filters from the last collection. Seems it doesn't clear state correctly.

**Pull request merged**